workers/unity/Packages/com.improbable.gdk.mobile/Editor/iOSLaunchUtils.cs

```diff
@@ -0,0 +1,292 @@
+                    Debug.LogError($"Was unable to read and modify {xcTestRunPath}.");
+                    return;
+                }
+                
+                if (useSimulator)
+                {
+                    EditorUtility.DisplayProgressBar("Launching Mobile Client", "Start iOS Simulator", 0.5f);
+
+                    // Start simulator
+                    if (RedirectedProcess.Command("xcrun")
+                        .WithArgs("instruments", "-w", deviceId, "-t", "Blank")
+                        .Run() != 0)
+                    {
+                        Debug.LogError("Was unable to start iOS Simulator.");
+                        return;
+                    }
+                }
+
+                EditorUtility.DisplayProgressBar("Launching Mobile Client", "Installing your app", 0.7f);
+
+                if (!TryLaunchApplication(deviceId, xcTestRunPath))
+                {
+                    Debug.LogError("Failed to start app on iOS device.");
+                }
+
+                EditorUtility.DisplayProgressBar("Launching Mobile Client", "Done", 1.0f);
+            }
+            finally
+            {
+                var traceDirectories = Directory
+                    .GetDirectories(Path.Combine(Application.dataPath, ".."), "*.trace")
+                    .Where(s => s.EndsWith(".trace"));
+                foreach (var directory in traceDirectories)
+                {
+                    Directory.Delete(directory, true);
+                }
+                
+                EditorUtility.ClearProgressBar();
+            }
+        }
+        
+        private static bool TryBuildXCodeProject(string developmentTeamId)
+        {
+            var process = RedirectedProcess.Command("xcodebuild")
+                .WithArgs("build-for-testing",
+                    "-project", Path.Combine(XCodeProjectPath, XCodeProjectFile),
+                    "-derivedDataPath", DerivedDataPath,
+                    "-scheme", "Unity-iPhone",
+                    $"DEVELOPMENT_TEAM={developmentTeamId}",
+                    "-allowProvisioningUpdates")
+                .Run();
+            
+            return process == 0;
+        }
+
+        private static bool TryLaunchApplication(string deviceId, string filePath)
+        {
+            var command = "osascript";
+            var commandArgs = $@"-e 'tell application ""Terminal""
+                                     activate
+                                     do script ""xcodebuild test-without-building -destination id={deviceId} -xctestrun {filePath}""
+                                     end tell'";
+
+            var processInfo = new ProcessStartInfo(command, commandArgs)
+            {
+                CreateNoWindow = false,
+                UseShellExecute = true,
+                WorkingDirectory = Common.SpatialProjectRootDir
+            };
+
+            var process = Process.Start(processInfo);
+
+            return process != null;
```

Comment:
Why not use `RedirectedProcess`?

Reply:
because we actually want to show the terminal as it contains any logs that a simulator or device may produce. don't think `RedirectedProcess` lets you do this and closing the terminal will also close the app on device or simulator

Reply:
Gotcha 👍

Reply:
Gotcha 👍